src/main/java/org/opensearch/security/httpclient/HttpClient.java

```diff
@@ -209,6 +205,30 @@ public HttpAsyncClientBuilder customizeHttpClient(HttpAsyncClientBuilder httpCli
         rclient = new RestHighLevelClient(builder);
     }
 
+    private HttpHost[] createHosts(String[] servers) {
+        /*
```

Comment:
Can we reuse an existing URL/URI parser, and then contruct the HttpHost via that process?

Reply:
@peternied I tried using URL and URI, but the tests didn't pass when URLs are like `127.0.0.1` or `[::1]:9200`.
Any suggestions?

Reply:
ChatGPT produced this [sample program](https://ideone.com/LkL4dF), and it looks like it works.  What didn't pass?

Reply:
Thanks a lot @peternied, here is what happened:
When creating the URI it needs the scheme, as you can see in the example you linked, so I fixed it, and it worked.
Then I removed the `ssl ? "https://" : "http://"` statement when creating the Host because it was already in the URL and it failed with `java.lang.AssertionError`.
Finally, I created the Host using URI parameters and it worked, however, when creating Hosts it doesn't add the path to the host `HttpHost(uri.getScheme(), uri.getHost(), uri.getPort());`, so 
`https://my-opensearch-cluster.company.com:9200/opensearch` is added as 
`https://my-opensearch-cluster.company.com:9200`
I couldn't find a way to add the path to the host, any ideas on this?

Reply:
Hmm, a path on a 'httphost' is a contradiction, then its not a `host` its an endpoint, or a host + a basepath.   Semantics aside, if you print out the list of transformed http hosts using the previous process do that have the path component?  Following up on that, are there tests that exercise that http client.  Because so long as those tests are still passing its not using that path segment a base path, so this approach would be safe.

If it is use the path as base path, then the data structure is wrong, then don't use http host, but create them as Node objects using the normalized url that was parsed.

Note; Also this seems more complex than originally thought so we might need more tests that properly validate this end to end kind of workflow where the client makes a network calls and we are sure they resolve correctly.

Thanks for digging in on this - let me know if you need help.

Reply:
Host class does not have _path_ component, but URI does.
There are not tests for URL with path, I was just trying to make it work based on this [comment](https://github.com/opensearch-project/security/issues/3321#issuecomment-1721290097).
I'll keep working on it to see what else I can find.

Reply:
From what I understand, the issue at hand seems to be of supporting urls, and the setting `plugins.security.audit.config.http_endpoints:  ` intuitively suggests setting the endpoint, i.e. `https://localhost:9200` and not host `localhost:9200`,  however it seems to be expecting a host `localhost:9200`. If that is the problem being solved, to support urls with custom opensearch endpoints, this change seems comprehensible to me. But, if that is not the case, then this might be more complicated then what it seems to be,

Reply:
Using the feedback I got, I created the last commit where it can create hosts from URLs and IP addresses, and I think it solved the main issue described in #3321